### DISCUSSION TITLE
Search panel improvements

### DISCUSSION THREAD

**Discussion by ghost** [mikekazakov/nimble-commander#436] [Ideas]
First of all, thanks a lot for putting so much effort into Nimble Commander. I just stumbled across it earlier today and already find it an invaluable tool for my everyday work. It'a real gem!

But I believe the "Find..." panel deserves a few improvements. It would be really great if one could sift through the search results, previewing items and opening paths in a file browser without having the search panel being closed immediately after the first interaction, and thereby losing all the results.

I could think of several ways of making the results panel a lot more valuable, for instance by...
- allowing commands such as "copy item path" in there as well
- allowing an item from the search results to be shown in a new window and thereby keeping the results open, either via contextual menu or by pressing opt while using the buttons at the bottom
- allowing the normal preview to be used in here as well. This could be particularly helpful when sifting through a collection of photos
- maybe opening the "Find" panel as a floating window instead of a sheet, to allow it being left open until being closed intentionally
- maybe opening the panel in a new tab? If that's technically possible anyway...

I'm sure there are a lot more possibilities here, but that's at least what I came up with from a mere user's point of view. But being a developer myself I realize that this could require a lot more effort than it would seem. So I ask you for your indulgence if these proposals are totally out of line 😇

**Comment by mikekazakov** [comment]
> First of all, thanks a lot for putting so much effort into Nimble Commander. I just stumbled across it earlier today and already find it an invaluable tool for my everyday work. It'a real gem!
 
Hey, thank you, glad you find NC useful!

> But I believe the "Find..." panel deserves a few improvements. It would be really great if one could sift through the search results, previewing items and opening paths in a file browser without having the search panel being closed immediately after the first interaction, and thereby losing all the results.

Doesn't panelisation already provide it?
i.e. clicking on the `Panel` button or pressing `Ctrl+P`.

> I could think of several ways of making the results panel a lot more valuable, for instance by...
> 
> * allowing commands such as "copy item path" in there as well
> * allowing an item from the search results to be shown in a new window and thereby keeping the results open, either via contextual menu or by pressing opt while using the buttons at the bottom
> * allowing the normal preview to be used in here as well. This could be particularly helpful when sifting through a collection of photos
> * maybe opening the "Find" panel as a floating window instead of a sheet, to allow it being left open until being closed intentionally
> * maybe opening the panel in a new tab? If that's technically possible anyway...

I think all of these should be technically doable.

> I'm sure there are a lot more possibilities here, but that's at least what I came up with from a mere user's point of view. But being a developer myself I realize that this could require a lot more effort than it would seem. So I ask you for your indulgence if these proposals are totally out of line 😇

Proposals and ideas are fine, as long as users also somehow contribute to the project.

**Comment by ghost** [comment]
Yes you're right in so far as the temporary panel indeed provides me with the search results in a way that allows me for example to preview the items. But the search sheet is gone, so I can't refine or modify my search but have to start over with all the search options being reset.

**Comment by Robert-M-Muench** [comment]
Things that I would like to see improved:
* Show the path to the search panel entry
* Return to prior view on ESC. I have very hard times to remember how to return to my previous view if it all.

The search result panel is a super useful thing, but currently it's enter only with no way out.
